lotus/models/qwen2_model.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Union
+
+import faiss
+import numpy as np
+import torch
+import torch.nn.functional as F
+from numpy.typing import NDArray
+from PIL import Image
+from tqdm import tqdm
+from transformers import Qwen2VLForConditionalGeneration, Qwen2VLProcessor
+
+from lotus.models.faiss_rm import FaissRM
+from lotus.types import RMOutput
+from lotus.utils import fetch_image
+
+
+class Qwen2Model(FaissRM):
```

Comment:
Is there a proper abstraction so we can support any VLM easily?

Reply:
Also let's add some tests for this if we want to keep it

Reply:
I will remove this for now. Will add it back with sem_sim_join